se/se_epub_generate_toc.py

```diff
@@ -412,17 +456,17 @@ def process_headings(soup: BeautifulSoup, textf: str, toc_list: list, nest_under
 		special_item = TocItem()
 		# Need to determine level depth.
 		# We don't have a heading, so get first content item
-		content_item = soup.find(["p", "header", "img"])
+		content_item = dom.xpath("//p | //header | //img")
```

Comment:
xpath should be able to return the closest parent section or article without needing to use `get_outer_sections`. Something like:

`//*[name() = 'p' or name() = 'header' or name = 'img']/ancestor::*[name() = 'section' or name() = 'article'][1]`

This xpath asks for:

1. Any element in (p | header | img)

2. Then, for the matching set, find any ancestor element of of those items that are (section | article). (Note that ancestor and parent are not the same: ancestor means any element that is the ancestor of the target node up to the root, while parent means the direct parent element of the target node and nothing else.)

3. Then, select the first instance of those matches (since a child node may have many ancestors up the tree, and we only want the first match)

I *think* this should work. If not, the concept should be possible with just a little tweaking.